### ISSUE TITLE
Can't get InputLayout to show error programmatically. 

### ISSUE THREAD

**Issue by galibimtiaz** [ZieIony/Carbon#278]
Not sure what is wrong but I can't show InputLayout error programmatically, 

When I call inputlayout.setError("Error"); it shows noting , but from xml carbon_pattern seems working fine.

**Comment by ZieIony** [comment]
InputLayout.setError(String) sets error text. To make error visible you need an invalid InputView inside.

Maybe there should be a way to make error visible manually.

**Comment by ZieIony** [comment]
I've added errorMode enum flag to InputLayout. It replaces errorEnabled boolean flag with three constants:
 - WhenInvalid - 'true' replacement
 - Never - 'false' replacement
 - Always - a new option for showing error message no matter what

The new flag can be used from xml (`carbon_errorMode`). InputLayout.setErrorEnabled(boolean) is now deprecated (but still works).

I also changed `carbon_errorMessage` to `carbon_error` to make that xml attribute consistent with the corresponding method. The old attribute is deprecated, but still works as well.
